lib/transmission_api/client.rb

```diff
@@ -54,15 +54,31 @@ def find(id)
     response["arguments"]["torrents"].first
   end
 
-  def create(filename)
+  def create(filename, download_dir = nil, bandwidthPriority = nil, paused = nil)
     log "add_torrent: #{filename}"
 
+    data = {
+          :filename => filename,
+	  :paused => paused,
+       } 
+
+    # add some of the optional parameters if provided
+    if (download_dir != nil)
```

Comment:
All of these can be turned into one liners and only check for truthy/falsy
```
data["download-dir"] = download_dir if download_dir
```
Another option (which would be my preference) would be to add all options regardless, then call `compact` on the hash to remove all nil values
```
option1 = 1
option2 = nil
{
  option1: option1,
  option2: option2,
}.compact # => { option1: 1 }

Reply:
Aha, that looks much nice, didn't realize there was a good way to do that.  I'll rewrite it to that since it's a much nicer way to do it.